extensions/VisualEditor/wikia/ApiAddMediaPermanent.php

```diff
@@ -41,7 +41,8 @@ private function executeImage() {
 			$file = new LocalFile( $title, RepoGroup::singleton()->getLocalRepo() );
 			$file->upload( $tempFile->getPath(), '', $pageText ? $pageText : '' );
 			return array(
-				'title' => $file->getTitle()->getText()
+				'title' => $file->getTitle()->getText(),
+				'url' => $file->getUrl()
```

Comment:
It would involve firing another request to MW to generate Title object based on file name and then generating URL for it. Or maybe you know a better way of doing this without firing another request? I was thinking about adding separate API method for mercury in MW so I don't have to change this one but there would be too much duplicated code and I decided not to proceed this way.